markdown/m2h/handlers/index.ts

```diff
@@ -86,18 +112,45 @@ export function buildLocalizedHandlers(locale: string): Handlers {
     blockquote(state: State, node: any): ReturnType<Handler> {
       const type = getNotecardType(node, locale);
       if (type) {
-        const isCallout = type == "callout";
-        if (isCallout) {
-          if (node.children[0].children.length <= 1) {
-            node.children.splice(0, 1);
-          } else {
-            node.children[0].children.splice(0, 1);
+        const isCallout = type.type == "callout";
+
+        if (type.isGFM) {
+          // Handle GFM proposed syntax
+          node.children[0].children[0].value =
+            node.children[0].children[0].value.replace(/\[!\w+\]\n/, "");
+
+          // If the type isn't a callout, add the magic keyword
+          if (!isCallout) {
+            node.children[0].children.unshift({
+              type: "strong",
+              children: [
+                {
+                  type: "text",
+                  value: type.magicKeyword,
+                },
+              ],
+            });
+            node.children[0].children[1].value =
+              (["zh-CN", "zh-TW"].includes(locale) ? "" : " ") +
+              node.children[0].children[1].value;
+          }
```

Comment:
This is what ensures that our formatting does not change, so that:

> [!NOTE]
> Foo bar

...renders as:

> **Note:** Foo bar

@yin1999 specifically stated that Chinese locales should not have a space as well.

Reply:
It looks like a separate change. We seem to currently render a space in Chinese, or am I missing something: https://developer.mozilla.org/zh-CN/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#%E7%A4%BA%E4%BE%8B

Reply:
Please see https://github.com/mdn/yari/pull/10168#discussion_r1551640392 for more details.

Reply:
Sorry, I missed to specify that I meant the special treatment for Chinese.

Reply:
Isn't the issue that the `msgstr`s for Chinese contain a trailing space? Could we remove that and just trim spaces above in line 72:

```diff
     // E.g. in en-US magicKeyword === Note:
-    const magicKeyword = grandChild.children[0].value;
+    const magicKeyword = grandChild.children[0].value.trim();
```

Reply:
The localization files do not have a trailing space, no -- if that were the case, I would simply remove them there!

The issue is that we are __adding__ a space in between the colon and the content of the noteblock, because that is what is grammatically correct for Western languages.  In the case of Chinese, however, adding a space is not proper grammar (as @yin1999 is indicating).  So, when a Chinese locale is in use, we don't want to add a space in between the magic word and the content.

Reply:
Gotcha, I was looking at the locale and saw a space, but it seems part of the Chinese colon symbol.